crates/core/src/db/datastore/locking_tx_datastore/btree_index.rs

```diff
@@ -29,39 +29,19 @@ use std::{
 /// [AlgebraicValue::I32(0)] = Row(ProductValue(...))
 /// [AlgebraicValue::Product(AlgebraicValue::I32(0), AlgebraicValue::I32(1))] = Row(ProductValue(...))
 /// ```
-#[derive(Clone, Eq, PartialEq, Ord, PartialOrd)]
-struct IndexKey {
-    value: AlgebraicValue,
-    row_id: RowId,
-}
-
-impl IndexKey {
-    #[tracing::instrument(skip_all)]
-    pub(crate) fn from_row(value: &AlgebraicValue, row_id: DataKey) -> Self {
-        Self {
-            value: value.clone(),
-            row_id: RowId(row_id),
-        }
-    }
-}
-
-pub struct BTreeIndexIter<'a> {
-    iter: btree_set::Iter<'a, IndexKey>,
-}
+type IndexKey = AlgebraicValue;
 
-impl Iterator for BTreeIndexIter<'_> {
-    type Item = RowId;
-
-    #[tracing::instrument(skip_all)]
-    fn next(&mut self) -> Option<Self::Item> {
-        self.iter.next().map(|key| key.row_id)
-    }
-}
+/// A sorted collection of row ids.
+///
+/// This is `Vec<_>` but we allow a single element to be stored inline
+/// to improve performance for the common case of one element.
```

Comment:
Extend docstring justifying why this needs to be sorted, and who maintains that.